test: add e2e tests for docs and tools pages

anushkaaaaaaaa opened the pull request:
<!--   Thank you for your contribution. Before you submit the pull request:
1. Follow our contribution guidelines
2. Test your changes and attach their results to the pull request.
3. Update the relevant documentation.
-->

**Description**

I have added E2E tests for the Docs Page and Tools Page in this PR using Cypress to ensure proper functionality and rendering. These tests cover key aspects such as-:
- Docs Page & Tools Page: Verifying critical user flows.

**Related issue(s)**
- #3671
<!-- If you refer to a particular issue, provide its number, otherwise, remove this section.
For example, `Resolves #123`, `Fixes #43`, or `See also #33`. The `See also #33` option will not automatically close the issue after the PR merge. -->

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Tests**
  * Added end-to-end suites for documentation and tools pages covering navigation, section/subsection checks, and card/link verification.
* **Tests / Helpers**
  * Added shared page-object helpers to streamline navigation, header and link verification across tests.
* **Tests / Fixtures**
  * Added structured fixtures for documentation sections and tooling resources to drive UI tests.
* **Chores**
  * Increased E2E command timeout and updated meeting and video entries in site configuration.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

anushkaaaaaaaa committed:
test: add e2e tests for docs and tools pages

anushkaaaaaaaa committed:
Merge branch 'asyncapi:master' into tools-docs

anushkaaaaaaaa committed:
fix: minor fixes

anushkaaaaaaaa committed:
Merge branch 'tools-docs' of https://github.com/anushkaaaaaaaa/website into tools-docs

anushkaaaaaaaa committed:
test: update e2e test files

anushkaaaaaaaa committed:
Merge branch 'master' into tools-docs

anushkaaaaaaaa committed:
test: small changes

anushkaaaaaaaa committed:
Merge branch 'tools-docs' of https://github.com/anushkaaaaaaaa/website into tools-docs

sambhavgupta0705 committed:
Merge branch 'master' into tools-docs

TRohit20 committed:
Merge branch 'master' into tools-docs

anushkaaaaaaaa committed:
Merge branch 'master' into tools-docs

anushkaaaaaaaa committed:
Merge branch 'master' into tools-docs

anushkaaaaaaaa committed:
fix: small changes

anushkaaaaaaaa committed:
Merge branch 'tools-docs' of https://github.com/anushkaaaaaaaa/website into tools-docs

anushkaaaaaaaa committed:
minor change

anushkaaaaaaaa committed:
minor chabge

anushkaaaaaaaa committed:
feat: implement base classes in e2e tests

anushkaaaaaaaa committed:
Merge branch 'master' into tools-docs

anushkaaaaaaaa committed:
Merge branch 'master' into tools-docs

anushkaaaaaaaa committed:
fix: enhance link verification methods

anushkaaaaaaaa committed:
Merge branch 'master' into tools-docs

anushkaaaaaaaa committed:
fix: update tools data import path in E2E tests

anushkaaaaaaaa committed:
Merge branch 'tools-docs' of https://github.com/anushkaaaaaaaa/website into tools-docs

anushkaaaaaaaa committed:
feat: add comprehensive tools data to toolsData.json

anushkaaaaaaaa committed:
Merge branch 'master' into tools-docs

anushkaaaaaaaa committed:
Merge branch 'master' into tools-docs

anshgoyalevil reviewed (dismissed):
Looks good. Lets ship it.

TRohit20 reviewed (dismissed):
Looks good to me, thank you and good job ! @anushkaaaaaaaa

anushkaaaaaaaa committed:
chore: remove Glee section from docsSections.json

anushkaaaaaaaa committed:
chore: format JSON structure in docsSections.json for consistency

anushkaaaaaaaa committed:
Merge branch 'master' into tools-docs

anushkaaaaaaaa committed:
Merge branch 'master' into tools-docs

sambhavgupta0705 commented on cypress/fixtures/toolsData.json:
```diff
@@ -0,0 +1,663 @@
+      { "href": "https://studio.asyncapi.com", "heading": "AsyncAPI Studio" }
+    ],
+    "GithubLinks": [
+      {
+        "href": "https://github.com/Shurtu-gal/action-test-bed",
+        "heading": "AsyncAPI Studio"
+      }
```
@anushkaaaaaaaa ptal

anushkaaaaaaaa committed:
chore: update tools and docs navigation methods in HomePage class

anushkaaaaaaaa committed:
Merge branch 'master' into tools-docs

sambhavgupta0705 committed:
Merge branch 'master' into tools-docs

anushkaaaaaaaa committed:
Merge branch 'master' into tools-docs

anushkaaaaaaaa committed:
Merge branch 'master' into tools-docs

sambhavgupta0705 committed:
Merge branch 'master' into tools-docs

sambhavgupta0705 commented:
@anushkaaaaaaaa did you have a look at coderabbit's comments

anushkaaaaaaaa committed:
Merge branch 'master' into tools-docs

anushkaaaaaaaa committed:
Merge branch 'tools-docs' of https://github.com/anushkaaaaaaaa/website into tools-docs

anushkaaaaaaaa committed:
refactor: replace verifyConceptSection with goToConceptsSection for consistency

anushkaaaaaaaa committed:
refactor: enhance page verification methods in ToolsPage

anushkaaaaaaaa committed:
refactor: update documentation links and headings in docsSections.json

anushkaaaaaaaa committed:
refactor: improve link verification logic in ToolsPage

anushkaaaaaaaa commented:
> @anushkaaaaaaaa did you have a look at coderabbit's comments

yes i did, but today after i updated the branch some new changes were added after which it generated some more comments which i have addressed as well

anushkaaaaaaaa committed:
Merge branch 'master' into tools-docs

sambhavgupta0705 committed:
Merge branch 'master' into tools-docs

anushkaaaaaaaa committed:
Add newline at end of meetings.json

Fix missing newline at end of meetings.json file.

anushkaaaaaaaa committed:
Fix JSON formatting in newsroom_videos.json

sambhavgupta0705 reviewed (dismissed):
LGTM!!

TRohit20 reviewed (dismissed):
LGTM!

anushkaaaaaaaa committed:
chore: add newline at the end of BasePageTools.js

anushkaaaaaaaa committed:
Merge branch 'tools-docs' of https://github.com/anushkaaaaaaaa/website into tools-docs

anushkaaaaaaaa committed:
docs: add Community Travel Funding guidelines

anushkaaaaaaaa committed:
docs: update links to AsyncAPI tool documentation

anushkaaaaaaaa committed:
docs: add manual tools building functionality and related tests

anushkaaaaaaaa committed:
remve line

anushkaaaaaaaa committed:
add line

anushkaaaaaaaa committed:
smhtg

sambhavgupta0705 reviewed (approved):
lgtm

TRohit20 reviewed (approved):
LGTM
